src/legacy/core_plugins/data/public/query/query_bar/lib/fetch_index_patterns.ts

```diff
@@ -40,15 +42,17 @@ export async function fetchIndexPatterns(indexPatternStrings: string[]) {
     return indexPatternStrings.includes(savedObject.attributes.title as string);
   });
 
+  const defaultIndex = uiSettings.get('defaultIndex');
```

Comment:
nit: not sure that it bring us any performance issues but looks like we need it this value only inside the fetchDefaultIndexPattern method. 

what about to pass `uiSettings `instead of `defaultIndex  `to `fetchDefaultIndexPattern ` method? 
or you can move declaration of fetchDefaultIndexPattern  to fetchIndexPatterns e.g. 

```
export async function fetchIndexPatterns(indexPatternStrings: string[],  uiSettings: UiSettingsClientContract) {
   const fetchDefaultIndexPattern = async (defaultIndex: string) => { ...}
...
}
```

Reply:
I'm actually going to get rid of this method in a subsequent PR.